### PR TITLE
[NEW] File upload

### PR DESCRIPTION
**Pull request by pranavpandey1998official**
@RocketChat/ReactNative

added support for file upload

ezgif com-video-to-gif (2)

WhatsApp Image 2019-05-14 at 3 28 11 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,8 +36,9 @@
     "react-native-action-sheet": "^2.2.0",
     "react-native-audio": "^4.3.0",
     "react-native-console-time-polyfill": "^1.2.1",
-    "react-native-device-info": "1.6.1",
-    "react-native-dialog": "^5.6.0",
+    "react-native-device-info": "^0.25.1",
+    "react-native-dialog": "^5.5.0",
```
Comment (djorkaeffalexandre) [inline package.json]: This downgrade is necessary?

File: app/i18n/locales/en.js [hunk 2]
```diff
@@ -118,6 +118,7 @@ export default {
 	Close_emoji_selector: 'Close emoji selector',
 	Choose: 'Choose',
 	Choose_from_library: 'Choose from library',
+	Choose_file: 'Choose file',
```
Comment (IlarionHalushka) [inline app/i18n/locales/en.js]: consider adding translations for other languages

File: app/containers/MessageBox/FilesActions.js [hunk 3]
```diff
@@ -26,6 +27,10 @@ export default class FilesActions extends PureComponent {
 		this.options.push(I18n.t('Choose_from_library'));
 		this.LIBRARY_INDEX = 2;
 
+		// File
+		this.options.push(I18n.t('Choose_file'));
```
Comment (IlarionHalushka) [inline app/containers/MessageBox/FilesActions.js]: I suggest to `this.options = [Cancel, TakePhoto...]` no need for all these `this.options.push`

**Comment by pranavpandey1998official** [comment]
> Not able to run on IOS: `null is not an object DocumentPicker.show` so created PR to your branch with fix: [pranavpandey1998official#1](https://github.com/pranavpandey1998official/Rocket.Chat.ReactNative/pull/1)

thanks man ,

File: app/containers/MessageBox/FilesActions.js (hunk 3 above)
Comment (pranavpandey1998official) [reply app/containers/MessageBox/FilesActions.js]: It's better readable with it1n

**Comment by djorkaeffalexandre** [comment]
When I try to upload a file happens this error:

File: app/containers/MessageBox/UploadModal.js [hunk 4]
```diff
@@ -63,18 +67,46 @@ const styles = StyleSheet.create({
...
+}))
 export default class UploadModal extends Component {
 	static propTypes = {
 		isVisible: PropTypes.bool,
 		file: PropTypes.object,
 		close: PropTypes.func,
 		submit: PropTypes.func,
-		window: PropTypes.object
+		window: PropTypes.object,
+		FileUpload_MediaTypeWhiteList: PropTypes.array,
```
Comment (djorkaeffalexandre) [inline app/containers/MessageBox/UploadModal.js]: Expected array but this value is a String.

Comment (pranavpandey1998official) [reply app/containers/MessageBox/UploadModal.js]: yes you are right : thansk:)

**Comment by djorkaeffalexandre** [comment]
Hi @pranavpandey1998official , I runned again your pr but have same error of my last comment, I searched about that and found this issue:
https://github.com/Elyx0/react-native-document-picker/issues/170

File: package.json [hunk 5]
```diff
@@ -39,6 +39,7 @@
     "react-native-console-time-polyfill": "^1.2.1",
     "react-native-device-info": "1.6.1",
     "react-native-dialog": "^5.6.0",
+    "react-native-document-picker": "Elyx0/react-native-document-picker#v3",
```
Comment (diegolmello) [inline package.json]: We don't want this experimental branch landing master: https://github.com/Elyx0/react-native-document-picker/tree/v3#warning-experimental-branch-of-v3-

Comment (pranavpandey1998official) [reply package.json]: I have some complications with the original , do you really want to use original one ???

**Comment by ccfiel** [comment]
When this be merge?

**Comment by djorkaeffalexandre** [comment]
@pranavpandey1998official I merged develop and add pt-BR translations in this pr:
https://github.com/pranavpandey1998official/Rocket.Chat.ReactNative/pull/5

File: ios/Podfile [hunk 6]
```diff
@@ -48,6 +48,8 @@ target 'RocketChatRN' do
 
   use_unimodules!
 
+  pod 'react-native-document-picker', :path => '../node_modules/react-native-document-picker'
```
[the remark this answers is not in the document]
Comment (diegolmello) [reply ios/Podfile]: I also noted you follow both `Automatically Link Native Modules` AND `CocoaPods`.
App wasn't compiling.
I fixed it.

[the remark this answers is not in the document]
Comment (pranavpandey1998official) [reply ios/Podfile]: thanx man

**Pull request merged**